### PR TITLE
Appointment read/search cancelled reason changes

### PR DESCRIPTION
**Pull request by SantanuBanerjee**
Description
----
Return Cancellation Reason in Appointment Read/Search

PR Checklist
----
Screenshot 2021-03-15 at 3 49 58 PM
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

**Comment by sujanb12** [comment]
Since we are adding a new field we need to list it in this file: https://github.com/cerner/fhir.cerner.com/blob/main/content/millennium/r4/scheduling/appointment.md

**Comment by SantanuBanerjee** [comment]
> Since we are adding a new field we need to list it in this file: https://github.com/cerner/fhir.cerner.com/blob/main/content/millennium/r4/scheduling/appointment.md
> 
> and update the appointment.yaml file to include the description if not present already.

New field added to the list :
https://github.com/cerner/fhir.cerner.com/commit/cea271830b2318b418286bdf3fea4b40657a0e68#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR35
but I guess appointment.yaml contains fields related to 'create' action, so it would not be relevant for this.

File: content/millennium/r4/scheduling/appointment.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ The following fields are returned if valued:
 
 * [Appointment id](http://hl7.org/fhir/R4/resource-definitions.html#Resource.id){:target="_blank"}
 * [Status](http://hl7.org/fhir/R4/appointment-definitions.html#Appointment.status){:target="_blank"}
+* [Cancellation Reason](http://hl7.org/fhir/R4/appointment-definitions.html#Appointment.cancelationReason){:target="_blank"}
```
Comment (sujanb12) [inline content/millennium/r4/scheduling/appointment.md:35]: Do we not need a terminology binding for this?

Comment (SantanuBanerjee) [reply content/millennium/r4/scheduling/appointment.md:35]: I am not sure about the terminology binding. I would need some opinion on this maybe.
@exopsRamba what do you think?

Comment (alexbeckner1) [reply content/millennium/r4/scheduling/appointment.md:35]: I think we need terminology binding as we are mapping millennium codeset to standard code system

Comment (SantanuBanerjee) [reply content/millennium/r4/scheduling/appointment.md:35]: Got it. Terminology bindings added, and snapshots added.
Screenshot 2021-03-18 at 5 54 31 PM
Screenshot 2021-03-18 at 5 54 51 PM
Screenshot 2021-03-18 at 5 55 14 PM


Changes:
https://github.com/cerner/fhir.cerner.com/commit/c44de0c25175498659d8a60f64949a635c699b17#diff-aedba7f37b773d7a314400db492774b8a9d406ed43aad2101793b82970106103R205

https://github.com/cerner/fhir.cerner.com/commit/c44de0c25175498659d8a60f64949a635c699b17#diff-aedba7f37b773d7a314400db492774b8a9d406ed43aad2101793b82970106103R607

https://github.com/cerner/fhir.cerner.com/commit/c44de0c25175498659d8a60f64949a635c699b17#diff-256db1c27cbe3136db6001a65a04bd3cffaf5beb61789a01f2c6da8b2676bc73R28

Comment (alexbeckner1) [reply content/millennium/r4/scheduling/appointment.md:35]: Do we need the fhir value set listed as well? Appointment cancellation reason -  https://www.hl7.org/fhir/r4/valueset-appointment-cancellation-reason.html

Comment (SantanuBanerjee) [reply content/millennium/r4/scheduling/appointment.md:35]: Done,
https://github.com/cerner/fhir.cerner.com/commit/c28a4715d5e50613d25066910c37d68feca097bd#diff-256db1c27cbe3136db6001a65a04bd3cffaf5beb61789a01f2c6da8b2676bc73R51

Comment (exopsRamba) [reply content/millennium/r4/scheduling/appointment.md:35]: Yeah, i believe it should match the system we are returning to match other resources: http://terminology.hl7.org/CodeSystem/appointment-cancellation-reason

Comment (SantanuBanerjee) [reply content/millennium/r4/scheduling/appointment.md:35]: Added
https://github.com/cerner/fhir.cerner.com/commit/c28a4715d5e50613d25066910c37d68feca097bd#diff-256db1c27cbe3136db6001a65a04bd3cffaf5beb61789a01f2c6da8b2676bc73R51

changed the system again.
https://github.com/cerner/fhir.cerner.com/commit/b55208c1d4716f3bb4b84ae86fe301aefaf1016e#diff-256db1c27cbe3136db6001a65a04bd3cffaf5beb61789a01f2c6da8b2676bc73R52

Comment (SantanuBanerjee) [reply content/millennium/r4/scheduling/appointment.md:35]: Screenshot 2021-03-18 at 8 01 06 PM

**Review by alexbeckner1 — APPROVED** [review]

File: content/millennium/r4/scheduling/appointment.md (hunk 1 above)
Comment (alexbeckner1) [reply content/millennium/r4/scheduling/appointment.md:35]: +1

**Review by sujanb12 — APPROVED** [review]

File: content/millennium/r4/proprietary-codes-and-systems.md [hunk 2]
```diff
@@ -601,6 +603,17 @@ This code set contains the area of the body the surgery procedure is being execu
       "display": "Left",
       "userSelected": true
     }
+    
+##### Code Set 14229 Scheduling Appointment Cancellation Reason
```
Comment (Coffeevee) [inline content/millennium/r4/proprietary-codes-and-systems.md:607]: Shouldn't this be `Appointment Cancelation Reason`?

Comment (SantanuBanerjee) [reply content/millennium/r4/proprietary-codes-and-systems.md:607]: Actually in all of the places there is "cancellation", which is why I am a little hesitant.
Aren't both the spellings grammatically correct?  @Coffeevee

File: content/millennium/r4/proprietary-codes-and-systems.md [hunk 3]
```diff
@@ -202,6 +202,8 @@ The contained Medication Resource supports proprietary codes for:
 
 The Appointment Resource supports proprietary codes for:
 
+* Appointment.cancelationReason codes are maintained in [Code Set 14229 Scheduling Appointment Cancellation Reason](#code-set-14229-scheduling-appointment-cancellation-reason)
+
```
Comment (exopsRamba) [inline content/millennium/r4/proprietary-codes-and-systems.md:206]: I'm not sure what happened but it looks like the second binding is missing again?

File: content/millennium/r4/proprietary-codes-and-systems.md (hunk 2 above)
Comment (Coffeevee) [reply content/millennium/r4/proprietary-codes-and-systems.md:607]: Yes, both spellings are technically gramatically correct, but I've been using 'cancelation reason' to stay consistent with the FHIR field

File: lib/resources/r4/appointment.yaml [hunk 4]
```diff
@@ -25,6 +25,36 @@ fields:
...
+        'text': 'Scheduled by mistake'
+      }
+    }
+  url: https://hl7.org/fhir/R4/appointment-definitions.html#Appointment.cancelationReason
+  binding:
+    description:  The coded reason for the appointment being cancelled.
+    terminology:
+      - display: Appointment cancellation reason
+        system: http://hl7.org/fhir/ValueSet/appointment-cancellation-reason
+        info_link: https://www.hl7.org/fhir/r4/valueset-appointment-cancellation-reason.html
```
Comment (exopsRamba) [inline lib/resources/r4/appointment.yaml:53]: Per my comment here: https://github.com/cerner/fhir.cerner.com/pull/584#discussion_r596925269

This should be: http://terminology.hl7.org/CodeSystem/appointment-cancellation-reason

File: content/millennium/r4/proprietary-codes-and-systems.md (hunk 2 above)
Comment (SantanuBanerjee) [reply content/millennium/r4/proprietary-codes-and-systems.md:607]: I can see in the hl7 documentation it is LL in many places ,
https://www.hl7.org/fhir/r4/valueset-appointment-cancellation-reason.html
wouldn't that be inconsistent as well?

File: lib/resources/r4/appointment.yaml (hunk 4 above)
Comment (SantanuBanerjee) [reply lib/resources/r4/appointment.yaml:53]: I had asked and verified the same with Alex and modified it the way he wanted.

File: content/millennium/r4/proprietary-codes-and-systems.md (hunk 2 above)
Comment (Coffeevee) [reply content/millennium/r4/proprietary-codes-and-systems.md:607]: Gotcha. Based on this and the fact that most people use the spelling with 'LL' anyways, I think I'm fine with leaving it this way then.

**Review by Coffeevee — APPROVED** [review]

File: lib/resources/r4/appointment.yaml (hunk 4 above)
Comment (SantanuBanerjee) [reply lib/resources/r4/appointment.yaml]: Talked to Alex again, and he is good with the one you suggested. Changed it.
https://github.com/cerner/fhir.cerner.com/commit/e800dbb7097efa29d479f05847ed49fd552dc444#diff-256db1c27cbe3136db6001a65a04bd3cffaf5beb61789a01f2c6da8b2676bc73R53

**Review by exopsRamba — APPROVED** [review]

**Review by ebbp123 — APPROVED** [review]

**Comment by BS062721** [comment]
Verified on live
Screen Shot 2021-03-18 at 3 35 01 PM

**Pull request merged**